### PR TITLE
Polylabel support

### PR DESCRIPTION
**Pull request by asyarb**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,13 +6,16 @@ API][arcgis-feature-service-rest-api].
...
 
-- [Features](#features)
-- [Install](#install)
-- [How to use](#how-to-use)
-- [How to query](#how-to-query)
-  - [Query Geometry](#query-geometry)
-  - [Query Properties](#query-properties)
-- [Site's `gatsby-node.js` example](#sites-gatsby-nodejs-example)
+- [gatsby-source-arcgis-feature-service](#gatsby-source-arcgis-feature-service)
+  - [Table of Contents](#table-of-contents)
```
Comment (angeloashmore) [inline README.md]: These headings shouldn't be part of the TOC.

Comment (angeloashmore) [inline README.md]: These headings shouldn't be part of the TOC.

File: README.md [hunk 2]
```diff
@@ -137,6 +140,25 @@ the ArcGIS Feature Service.
 }
 ```
 
+### Query Polylabel
+
+Polylabel data (see: [polylabel](https://github.com/mapbox/polylabel)) is
+provided on the `polylabel` field.
+
+`polylabel` is a longitude, latitude tuple of the most distant, internal point
+from the polygon outline.
```
Comment (angeloashmore) [inline README.md]: Could this read something like this? Something less technical sounding:

> Polylabel data is provided on the `polylabel` field. A polylabel is the optimal point within a polygon to place a marker provided as a `[x, y]` pair.
> 
> See [Mapbox's official Polylabel documentation][polylabel] for more details.

File: src/gatsby-node.js [hunk 3]
```diff
@@ -29,6 +30,7 @@ export const sourceNodes = async (gatsbyContext, pluginOptions) => {
   createTypes(`
     type ${FEATURE_TYPE} implements Node {
       geometry: JSON!
+      raw: JSON!
```
Comment (angeloashmore) [inline src/gatsby-node.js]: `raw` doesn't appear to be provided to the node in `createNode`?

File: src/gatsby-node.js [hunk 4]
```diff
@@ -38,6 +40,7 @@ export const sourceNodes = async (gatsbyContext, pluginOptions) => {
       ...feature,
       id: createNodeId([name, feature?.id].filter(Boolean).join(' ')),
       featureId: feature?.id,
+      polylabel: polylabel(feature.geometry.coordinates, 0.1),
```
Comment (angeloashmore) [inline src/gatsby-node.js]: Can you check the GeoJSON spec to ensure `geometry.coordinates` is always provided? What happens when the feature is a line? Or a point? Does polylabel work for those GeoJSON types?

GeoJSON RFP here: https://tools.ietf.org/html/rfc7946

**Review by angeloashmore — CHANGES_REQUESTED** [review]
Looks good! Left a few notes. Mainly just need to make sure the new polylabel stuff handles non-polygon GeoJSON types.

File: src/gatsby-node.js (hunk 4 above)
Comment (asyarb) [reply src/gatsby-node.js]: I believe any GeoJSON type has `geometry.coordinates`.

See section 3.1 in the linked RFP: https://tools.ietf.org/html/rfc7946#section-3.1

> A GeoJSON Geometry object of any type other than 
> "GeometryCollection" has a member with the name "coordinates".

File: src/gatsby-node.js (hunk 3 above)
Comment (asyarb) [reply src/gatsby-node.js]: You're right, that's probably left over I forgot to delete.

File: README.md (hunk 1 above)
Comment (asyarb) [reply README.md]: Yeah, probably autogenerated from my plugin. Removed.

File: README.md (hunk 2 above)
Comment (asyarb) [reply README.md]: Yep. Revised.

Comment (angeloashmore) [reply README.md]: I think something happened in the copy/paste here. Can you double check?

Also the link at the end of the doc is in the wrong format. Should be `[name]: url` format.

File: src/gatsby-node.js (hunk 4 above)
Comment (asyarb) [reply src/gatsby-node.js]: Yeah, just tested on lines and points, and it wont work on those types. I've changed to only provide polylabel if the feature geometry type is a `Polygon`

File: README.md (hunk 2 above)
Comment (asyarb) [reply README.md]: My b on the link format, but the above should be fixed now hopefully, lol

**Pull request merged**